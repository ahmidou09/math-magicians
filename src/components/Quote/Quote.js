import React, { useState, useEffect } from 'react';
import axios from 'axios';

const Quote = () => {
  const [quote, setQuote] = useState('');
  const [loading, setLoading] = useState(true);
  const [error, setError] = useState(null);

  useEffect(() => {
    const category = 'love';
    const apiKey = 'YFsFy9KojEgPBCbeWzF6nw==6qthtK3zADwmr5pz';

    axios.get(`https://api.api-ninjas.com/v1/quotes?category=${category}`, {
      headers: {
        'X-Api-Key': apiKey,
        'Content-Type': 'application/json',
      },
    })
      .then((response) => {
        if (response.data.length > 0) {
          setQuote(() => response.data[0].quote);
        } else {
          setError('No quotes found for the specified category.');
        }
        setLoading(false);
      })
      .catch((err) => {
        setError(`Error fetching quote: ${err.message}`);
        setLoading(false);
      });
  }, []);

  return (
    <div className="quote">
      {loading ? <p>Loading...</p> : null}
      {error ? <p>{error}</p> : null}
      {quote && !loading && !error ? <p>{quote}</p> : null}
    </div>
  );
};

export default Quote;
