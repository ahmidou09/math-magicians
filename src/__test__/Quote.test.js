import React from 'react';
import { render } from '@testing-library/react';
import Quote from '../components/Quote/Quote';

it('renders loading state initially', () => {
  const { asFragment } = render(<Quote />);
  expect(asFragment()).toMatchSnapshot();
});
