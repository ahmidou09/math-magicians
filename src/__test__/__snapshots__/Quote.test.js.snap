// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders loading state initially 1`] = `
<DocumentFragment>
  <div
    class="quote"
  >
    <p>
      Loading...
    </p>
  </div>
</DocumentFragment>
`;
